### ISSUE TITLE
r/aws_dx_gateway_association: Allowed prefix support

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#8199]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

Direct Connect gateway associations now support configuration of allowed prefixes.
This functionality was added in [AWS SDK v1.19.4](https://github.com/aws/aws-sdk-go/releases/tag/v1.19.4):

> ... This release also enables Direct Connect Gateway owners to allocate allowed prefixes from each associated VPCs.

Related:
* https://github.com/terraform-providers/terraform-provider-aws/issues/2140
* https://github.com/terraform-providers/terraform-provider-aws/pull/4896
* https://github.com/terraform-providers/terraform-provider-aws/issues/8100
* https://github.com/terraform-providers/terraform-provider-aws/pull/8108

**Comment by ewbankkit** [comment]
Proposed syntax:

```hcl
resource "aws_dx_gateway_association" "example" {
  dx_gateway_id  = "${aws_dx_gateway.example.id}"
  vpn_gateway_id = "${aws_vpn_gateway.example.id}"

  allowed_prefix {
    cidr = "11.0.0.0/16"
  }

  allowed_prefix {
    cidr = "12.0.0.0/16"
  }
}
```

**Comment by bflad** [comment]
Support for this on the existing `aws_dx_gateway_association` resource has been merged and will release with version 2.8.0 of the Terraform AWS Provider, likely later this week.

**Comment by nywilken** [comment]
This has been released in [version 2.8.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#280-april-26-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
